src/ec/eddsa.rs

```diff
@@ -297,60 +273,80 @@ fn elem_sq_in_place(a: &mut Elem) {
 #[doc(hidden)]
 #[no_mangle]
 pub unsafe extern fn eddsa_elem_invert(out: &mut Elem, a: &Elem) {
+    // Sets |out| to be |a| * |b| (mod p).
+    fn mul(out: &mut Elem, a: &Elem, b: &Elem) {
+        unsafe { GFp_fe_mul(out.as_mut_ptr(), a.as_ptr(), b.as_ptr()); }
+    }
+
+    // Sets |a| to be |a| * |b| (mod p).
+    fn mul_in_place(a: &mut Elem, b: &Elem) {
+        unsafe { GFp_fe_mul(a.as_mut_ptr(), a.as_ptr(), b.as_ptr()); }
+    }
+
+    // Sets |out| to be |a|**2 (mod p).
+    fn sqr(out: &mut Elem, a: &Elem) {
+        unsafe { GFp_fe_sq(out.as_mut_ptr(), a.as_ptr()); }
+    }
+
+    // Sets |a| to be |a|**2 (mod p).
+    fn sqr_in_place(a: &mut Elem) {
+        unsafe { GFp_fe_sq(a.as_mut_ptr(), a.as_ptr()); }
+    }
+
     let mut t0 = [0i32; ELEM_LIMBS];
     let mut t1 = [0i32; ELEM_LIMBS];
     let mut t2 = [0i32; ELEM_LIMBS];
     let mut t3 = [0i32; ELEM_LIMBS];
 
-    elem_sq(&mut t0, a);
-    elem_sq(&mut t1, &t0);
+    sqr(&mut t0, a);
+    sqr(&mut t1, &t0);
     for _ in 1..2 {
-        elem_sq_in_place(&mut t1);
+        sqr_in_place(&mut t1);
     }
-    elem_mul_in_place(&mut t1, a);
-    elem_mul_in_place(&mut t0, &t1);
-    elem_sq(&mut t2, &t0);
-    elem_mul_in_place(&mut t1, &t2);
-    elem_sq(&mut t2, &t1);
+    mul_in_place(&mut t1, a);
```

Comment:
Notice that every loop of `sqr_in_place()` is followed by `mul_in_place()` (Including the last one, because P is odd and so P - 2 is odd). That, and improved readability are why I did this:

https://github.com/briansmith/ring/blob/86a45d046e92640463923b2e35c2a94387109789/src/ec/suite_b/ops/p256.rs#L86-L94
https://github.com/briansmith/ring/blob/86a45d046e92640463923b2e35c2a94387109789/src/ec/suite_b/ops/ops.rs#L354-L373

Let's do the same here. (Note I guess I used `acc` instead of `r` there; I guess I could go either way as long as everything is made consistent.)